sql/core/src/main/scala/org/apache/spark/sql/functions.scala

```diff
@@ -358,9 +358,10 @@ object functions {
    * @since 1.3.0
    */
   @scala.annotation.varargs
-  def countDistinct(expr: Column, exprs: Column*): Column = {
-    withAggregateFunction(Count.apply((expr +: exprs).map(_.expr)), isDistinct = true)
-  }
+  def countDistinct(expr: Column, exprs: Column*): Column =
+    // For usage like countDistinct("*"), we should let analyzer expand star and
```

Comment:
how do we handle it in `count`?

Reply:
`count("*")` doesn't go into Analyzer. It is transformed to `count(1)` in `AstBuilder` and `functions` before analysis.

https://github.com/apache/spark/blob/0407070945fc090c84dde7a84e1b24a1acb131d4/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/AstBuilder.scala#L1363-L1370

https://github.com/apache/spark/blob/90085a184797f8bddbff8ca6ec7a60f3899c1a86/sql/core/src/main/scala/org/apache/spark/sql/functions.scala#L337-L343